### PR TITLE
sys/posix/pthread/pthread.c: fix pthread reaper

### PR DESCRIPTION
**Pull request by JulianHolzwarth**
### Contribution description
Fixing the pthread reaper. The reaper takes less memory (same stacksize as Idle thread) and pthreads are created sleeping instead of `THREAD_CREATE_WOUT_YIELD`.

### Testing procedure
Enable Debug in `sys/posix/pthread/pthread.c` and run `BOARD=native make -C tests/pthread/  flash test`
after fix:
```
main(): This is RIOT! (Version: 2019.10-devel-934-ga4d0c-pr/posix/pthread/pthread_reaper_fix)
main: parameter = 6
pthread: parameter = 6
pthread: factorial = 720
pthread_exit(0x2d0), self == 0x57e87008
pthread_reaper(): free(0x57e87038)
main: factorial = 720
SUCCESS
```

current output:
```
main(): This is RIOT! (Version: 2019.10-devel-958-g04b926)
main: parameter = 6
pthread: parameter = 6
pthread: factorial = 720
pthread_exit(0x2d0), self == 0x57c02008
main: factorial = 720
SUCCESS

```

### CODE REVIEW

**Comment by JulianHolzwarth** [comment]
@smlng Could you look at this?

**Comment by smlng** [comment]
I'm unsure on the what and why, can you please elaborate what you fix and why it is necessary, THX!

File: sys/posix/pthread/pthread.c [hunk 1]
```diff
@@ -38,10 +38,10 @@
 #define ENABLE_DEBUG (0)
 
 #if ENABLE_DEBUG
-#   define PTHREAD_REAPER_STACKSIZE THREAD_STACKSIZE_MAIN
+#   define PTHREAD_REAPER_STACKSIZE (THREAD_STACKSIZE_IDLE + THREAD_EXTRA_STACKSIZE_PRINTF)
```
Comment (miri64) [inline sys/posix/pthread/pthread.c]: Have you tested, if this is enough?

Comment (JulianHolzwarth) [reply sys/posix/pthread/pthread.c]: We found an issue on msba2. Fixing it now.

**Comment by JulianHolzwarth** [comment]
@smlng Before this pr the pthread_reaper (thread that frees stack memory of pthreads) was never created. After this fix the pthread_reaper is correctly created ( checking the output of `thread_create`  and handling errors)
Also reducing stacksize of the pthread_reaper.

File: sys/posix/pthread/pthread.c (hunk 1 above)
Comment (JulianHolzwarth) [reply sys/posix/pthread/pthread.c]: It should be enough now.

**Comment by JulianHolzwarth** [comment]
the output of `ps` after thread has been freed for some tested boards (tested with @cladmi )
[ps_output.arduino-mega2560.after.txt](https://github.com/RIOT-OS/RIOT/files/3653463/ps_output.arduino-mega2560.after.txt)
[ps_output.cc2650-launchpad.after.txt](https://github.com/RIOT-OS/RIOT/files/3653464/ps_output.cc2650-launchpad.after.txt)
[ps_output.ek-lm4f120xl.after.txt](https://github.com/RIOT-OS/RIOT/files/3653465/ps_output.ek-lm4f120xl.after.txt)
[ps_output.esp32-wroom-32.after.txt](https://github.com/RIOT-OS/RIOT/files/3653466/ps_output.esp32-wroom-32.after.txt)
[ps_output.frdm-k64f.after.txt](https://github.com/RIOT-OS/RIOT/files/3653467/ps_output.frdm-k64f.after.txt)
[ps_output.frdm-kw41z.after.txt](https://github.com/RIOT-OS/RIOT/files/3653468/ps_output.frdm-kw41z.after.txt)
[ps_output.msba2.after.txt](https://github.com/RIOT-OS/RIOT/files/3653469/ps_output.msba2.after.txt)
[ps_output.mulle.after.txt](https://github.com/RIOT-OS/RIOT/files/3653470/ps_output.mulle.after.txt)
[ps_output.nrf52dk.after.txt](https://github.com/RIOT-OS/RIOT/files/3653471/ps_output.nrf52dk.after.txt)
[ps_output.nucleo-f103rb.after.txt](https://github.com/RIOT-OS/RIOT/files/3653472/ps_output.nucleo-f103rb.after.txt)
[ps_output.pba-d-01-kw2x.after.txt](https://github.com/RIOT-OS/RIOT/files/3653473/ps_output.pba-d-01-kw2x.after.txt)
[ps_output.sltb001a.after.txt](https://github.com/RIOT-OS/RIOT/files/3653474/ps_output.sltb001a.after.txt)
[ps_output.stm32f3discovery.after.txt](https://github.com/RIOT-OS/RIOT/files/3653475/ps_output.stm32f3discovery.after.txt)
[ps_output.wsn430-v1_3b.after.txt](https://github.com/RIOT-OS/RIOT/files/3653476/ps_output.wsn430-v1_3b.after.txt)

**Pull request merged**